src/Sources/Code/Module.cpp

```diff
@@ -234,71 +276,125 @@ void WriteSimVar(SimVar& simVar, Client* client) {
-			var1.ID,
-			var1.Offset,
-			sizeof(float),
-			0
-		);
 
-		client->MaxClientDataDefinition = SimVars->size();
+	if(isString) {
+		newStringSimVar.Name = code;
+		newStringSimVar.ID = StringSimVars->size() + client->DataDefinitionIdStringVarsStart;
+		newStringSimVar.Offset = StringSimVars->size() * MOBIFLIGHT_STRING_SIMVAR_VALUE_MAX_LEN;
+		newStringSimVar.Value.empty();
+		StringSimVars->push_back(newStringSimVar);
+	} else {
+		newSimVar.Name = code;
+		newSimVar.ID = SimVars->size() + client->DataDefinitionIdSimVarsStart;
+		newSimVar.Offset = SimVars->size() * (sizeof(float));
+		newSimVar.Value = 0.0F;
+		SimVars->push_back(newSimVar);
 	}
+
+
+	if (client->MaxClientDataDefinition < (SimVars->size() + StringSimVars->size())) {
+		if(isString) {
+			hr = SimConnect_AddToClientDataDefinition(
+				g_hSimConnect,
+				newStringSimVar.ID,
+				newStringSimVar.Offset,
+				MOBIFLIGHT_STRING_SIMVAR_VALUE_MAX_LEN,
+				0
+			);
+
+			if (hr != S_OK) {
+				fprintf(stderr, "MobiFlight[%s]: Error on adding Client Data \"%s\" with String-ID: %u, String-Offset: %u and Size: %u\n", client->Name.c_str(), newStringSimVar.Name.c_str(), newStringSimVar.ID, newStringSimVar.Offset, MOBIFLIGHT_STRING_SIMVAR_VALUE_MAX_LEN);
+			}
 #if _DEBUG
-	std::cout << "MobiFlight[" << client->Name.c_str() << "]: RegisterSimVar SimVars Size: " << SimVars->size() << std::endl;
+			else {
+				std::cout << "MobiFlight[" << client->Name.c_str() << "]: Added String-SimVar > " << newStringSimVar.Name.c_str();
+				std::cout << " with String-ID: " << newStringSimVar.ID << ", String-Offset: " << newStringSimVar.Offset << " and Size: " << MOBIFLIGHT_STRING_SIMVAR_VALUE_MAX_LEN << std::endl;
+			}
+			std::cout << "MobiFlight[" << client->Name.c_str() << "]: RegisterSimVar StringSimVars Size: " << StringSimVars->size() << std::endl;
 #endif
+		}
+		else {
+			hr = SimConnect_AddToClientDataDefinition(
+				g_hSimConnect,
+				newSimVar.ID,
+				newSimVar.Offset,
+				sizeof(float),
+				0
+			);
+
+			if (hr != S_OK) {
+				fprintf(stderr, "MobiFlight[%s]: Error on adding Client Data \"%s\" with ID: %u, Offset: %u and Size: %lu\n", client->Name.c_str(), newSimVar.Name.c_str(), newSimVar.ID, newSimVar.Offset, sizeof(float));
+			}
+#if _DEBUG
+			else {
+				std::cout << "MobiFlight[" << client->Name.c_str() << "]: Added SimVar > " << newSimVar.Name.c_str();
+				std::cout << " with ID: " << newSimVar.ID << ", Offset: " << newSimVar.Offset << " and Size: " << sizeof(float) << std::endl;
+			}
+			std::cout << "MobiFlight[" << client->Name.c_str() << "]: RegisterSimVar SimVars Size: " << SimVars->size() << std::endl;
+#endif
+		}
+		client->MaxClientDataDefinition = (SimVars->size() + StringSimVars->size());
+	}
 
-	FLOAT64 val;
-	WriteSimVar(var1, client);
-
-	execute_calculator_code(std::string(code).c_str(), &val, NULL, NULL);
-	var1.Value = val;
-	
-	WriteSimVar(var1, client);
 
+	if(isString) {
+		PCSTRINGZ charVal = nullptr;
+		WriteSimVar(newStringSimVar, client);
```

Comment:
I would propose now to really remove that (unecessary?) first WriteSimVar. 
@DocMoebiuz  this was there from the beginning, if you cannot remember any reason we should remove it now. Currently each float variable will get an initial 0.0 callback before the real value.